src/validations.test.js

```diff
@@ -463,4 +464,225 @@ describe('validations', () => {
       expect(result).toEqual({ charLength: true, letter: true, digit: true });
     });
   });
+
+  describe('validateName', () => {
+    it('should accept an empty name', () => {
+      // Arrange
+      const name = '';
+
+      // Act
+      const result = validateName(name);
+
+      // Assert
+      expect(result).toBeNull();
+    });
+
+    it('should not accept a name with one char', () => {
+      // Arrange
+      const name = 'a';
+
+      // Act
+      const result = validateName(name);
+
+      // Assert
+      expect(result).toEqual('validation_messages.error_invalid_name');
+    });
+
+    it('should accept a name with two chars', () => {
+      // Arrange
+      const name = 'Fe';
+
+      // Act
+      const result = validateName(name);
+
+      // Assert
+      expect(result).toBeNull();
+    });
+
+    it('should accept a name with more than one char in chinese language', () => {
+      // Arrange
+      const name = '陸軍上校';
+
+      // Act
+      const result = validateName(name);
+
+      // Assert
+      expect(result).toBeNull();
+    });
+
+    it('should accept a name with point', () => {
+      // Arrange
+      const name = 'A.';
+
+      // Act
+      const result = validateName(name);
+
+      // Assert
+      expect(result).toBeNull();
+    });
+
+    it('should accept a name with spaces', () => {
+      // Arrange
+      const name = 'John Snow';
+
+      // Act
+      const result = validateName(name);
+
+      // Assert
+      expect(result).toBeNull();
+    });
+
+    it('should accept names with single quotes', () => {
+      // Arrange
+      const name = "D'onofrio";
+
+      // Act
+      const result = validateName(name);
+
+      // Assert
+      expect(result).toBeNull();
+    });
+
+    it('should not accept names starting with apostrophe', () => {
+      // Arrange
+      const name = '`A';
+
+      // Act
+      const result = validateName(name);
+
+      // Assert
+      expect(result).toEqual('validation_messages.error_invalid_name');
+    });
+
+    it('should not accept names with emojis', () => {
+      // Arrange
+      const name = 'a😀';
+
+      // Act
+      const result = validateName(name);
+
+      // Assert
+      expect(result).toEqual('validation_messages.error_invalid_name');
+    });
+
+    it('should not accept names starting with single quotes', () => {
+      // Arrange
+      const name = "'A";
+
+      // Act
+      const result = validateName(name);
+
+      // Assert
+      expect(result).toEqual('validation_messages.error_invalid_name');
+    });
+
+    it('should not accept a name with point in the beginning', () => {
+      // Arrange
+      const name = '.A';
+
+      // Act
+      const result = validateName(name);
+
+      // Assert
+      expect(result).toEqual('validation_messages.error_invalid_name');
+    });
+
+    it('should not accept a name with special character', () => {
+      // Arrange
+      const name = 'Joh=n';
```

Comment:
What do you think about moving this scenario to `should not accept a name with special character` (in the array)